packages/mongodb-chatbot-server/src/utils.ts

```diff
@@ -53,7 +53,9 @@ export const sendErrorResponse = ({
     message: stripIndent`Responding with ${httpStatus} status and error message: ${errorMessage}.
     ${errorDetails ? `Error details: ${errorDetails}` : ""}`,
   });
-  return res.status(httpStatus).json({ error: errorMessage });
+  if (!res.writableEnded) {
+    return res.status(httpStatus).json({ error: errorMessage });
+  }
```

Comment:
what happens if there's an error and the writeable has ended? 

also are you sure this behavior holds for when the server isn't streaming responses?

Reply:
This case probably _shouldn't_ happen if we structure things right, but it's a defensive check. If `writableEnded` is `true` then we've already finished responding and can't send anything else - the `.json()` call will throw an error.